core/libraries/batch/JobHandlers/RegistrationsReport.php

```diff
@@ -244,17 +241,17 @@ public function continue_job(JobParameters $job_parameters, $batch_size = 50)
      * @param int $limit
      * @param array $question_labels the IDs for all the questions which were answered by someone in this selection
      * @param array $query_params for using where querying the model
-     * @param bool $incude_checkins
+     * @param int|null $DTT_id
      * @return array top-level keys are numeric, next-level keys are column headers
      * @throws \EE_Error
      * @throws \ReflectionException
      */
-    public function get_csv_data_for($event_id, $offset, $limit, $question_labels, $query_params, $incude_checkins = false)
+    public function get_csv_data_for($event_id, $offset, $limit, $question_labels, $query_params, $DTT_id = null)
```

Comment:
plz try to maintain type consistency whenever possible. So since `$DTT_id` is an integer, it would be better to use an integer for the default value. Now that said, this doesn't always work and there are some cases where you need to be able to determine when a value has simply not been set, and need to use a value that can not be misinterpreted as a valid value. 

For example, using `0` as a default value for a quantity may not be good in cases where you only want to apply a change when a valid new value is provided. Since zero is a potentially valid value for a quantity, it's impossible to know whether that value was simply not set, or whether the user really wants to set the quantity to zero. In that case, `null` may work better. 

But for the most part, plz try to avoid type switching as much as possible as it greatly increases the chances of introducing issues that are tough to troubleshoot.

Oh and can you plz change `$DTT_id` to `$DTT_ID`?

Reply:
Yes, another good point. Thanks for sharing it.